### PR TITLE
Xm dev implemented the hex-tool

### PR DESCRIPTION
**Pull request by ju-liu**

### CODE REVIEW

**Review by Goldeninscriptions — APPROVED** [review]

**Review by Syj-0525 — COMMENTED** [review]
There are no issues after the review.

**Review by Goldeninscriptions — APPROVED** [review]
手动解决了conflict

File: src/Mesh/Gmsh_FileIO.cpp [hunk 1]
```diff
@@ -1398,15 +1395,15 @@ void Gmsh_FileIO::write_quadratic_sur_vtu( const std::string &vtu_filename,
...
+    ele_3d += static_cast<std::string>("tetrahedron");
   }
   else if (nlocbas_2d == 9 && nlocbas_3d == 27)
   {
     nVertex_2d = 4;
     nVertex_3d = 8;
-    ele_2d = static_cast<std::string>("quadrilateral");
-    ele_3d = static_cast<std::string>("hexahedron");
+    ele_2d += static_cast<std::string>("quadrilateral");
+    ele_3d += static_cast<std::string>("hexahedron");
```
Comment (ju-liu) [inline src/Mesh/Gmsh_FileIO.cpp]: ？

File: src/Mesh/Gmsh_FileIO.cpp [hunk 2]
```diff
@@ -424,13 +422,13 @@ void Gmsh_FileIO::write_vtp( const std::string &vtp_filename,
...
-    ele_3d = static_cast<std::string>("tetrahedron");
+    ele_2d += static_cast<std::string>("triangle");
+    ele_3d += static_cast<std::string>("tetrahedron");
   }
   else if (nlocbas_2d == 4 && nlocbas_3d == 8)
   {
-    ele_2d = static_cast<std::string>("quadrilateral");
-    ele_3d = static_cast<std::string>("hexahedron");
+    ele_2d += static_cast<std::string>("quadrilateral");
+    ele_3d += static_cast<std::string>("hexahedron");
```
Comment (ju-liu) [inline src/Mesh/Gmsh_FileIO.cpp]: ？

Comment (Goldeninscriptions) [reply src/Mesh/Gmsh_FileIO.cpp]: 刚刚没pull， 调回去了

**Pull request merged**